pkg/mysql/rows.go

```diff
@@ -114,6 +114,45 @@ func (row *Row) GetColumnValue(column string) (interface{}, error) {
 	return nil, nil
 }
 
+func (rows *TextRow) Encode(row []*proto.Value, fields []proto.Field, columnNames []string) proto.Row {
+	var val []byte
+	var isNull bool
+	var err error
+
+	for i := 0; i < len(fields); i++ {
+		field := fields[i].(*Field)
+		val = append(val, appendLengthEncodedInteger(row[i].Raw, uint64(row[i].Len))...)
+		if err == nil {
+			if !isNull {
+				switch field.fieldType {
+				case mysql.FieldTypeTimestamp, mysql.FieldTypeDateTime,
+					mysql.FieldTypeDate, mysql.FieldTypeNewDate:
+					data, err := appendDateTime(row[i].Raw, row[i].Val.(time.Time))
+					if err != nil {
+						return nil
+					}
+					val = append(val, data...)
+					if err == nil {
+						continue
+					}
+				default:
+					continue
+				}
+			} else {
+				continue
+			}
+		}
+	}
```

Comment:
Need to refactor this block, too many 'continue'.

Reply:
address